test/built-ins/Temporal/PlainDateTime/prototype/withPlainDate/argument-string-non-iso-calendar.js

```diff
@@ -0,0 +1,23 @@
+// Copyright (C) 2022 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-temporal.plaindatetime.prototype.withplaindate
+description: Switching to a non-ISO calendar is acceptable
+features: [Temporal]
+---*/
+
+const dt = new Temporal.PlainDateTime(1995, 12, 7, 3, 24, 30);
+const shifted = dt.withCalendar("japanese").withPlainDate("2008-09-06");
```

Comment:
This will fail on non-ECMA-402 hosts, so it should use a custom calendar.

Reply:
done